### ISSUE TITLE
How to prune self-defined models?

### ISSUE THREAD

**Issue by smxzxcghl** [NervanaSystems/distiller#460]
Hello,
Thank you for the awesome library.It is quite useful for me.
I have a question about adding self-defined models. I found all the classification tasks use compress_classification.py as main function while object detection ones use compress detector.py. If I prune a model of other tasks such as segmentation, is it more efficient to edit new scripts or just add models and datasets in the workflow of compress_classification.py?

The question really confuses me.Thank you.

**Comment by nzmora** [comment]
Hi @smxzxcghl 

Distiller is a library, and also comes with a bunch of examples for compressing models of different __tasks__.  Most compression algorithms (with the exception of post-training-quantization) involve some form of training (sometimes referred to as "model retraining" or "model fine-tuning").  I wrote __tasks__ in bold because this is a crucial point: different tasks train differently.  Although the main training logic of different task is shared (forward() to compute the loss; backward() to compute the gradients; update the weights), there are differences in the control-flow of image-classification and language-model learning.  Even between image-classification and object-detection there are lots of differences.  And this is why you see different examples in our code base.  
While we were writing Distiller, most of the research papers in model-compression used image-classifiers, and that's why you see an unbalanced number of examples using `compress_classification.py`.  
If you want to try compressing new image-classifiers try using `compress_classification.py`, but if you want to try image-segmentation tasks, for example, then you'll need to add support for Distiller in your image-segmentation code.  We tried to [explain how to do this](https://nervanasystems.github.io/distiller/tutorial-lang_model.html) using a language-model task.

Cheers
Neta
